### PR TITLE
[patch] Correct support versions of IBM App Connect

### PR DESCRIPTION
**Pull request by andrercm**
Current default version of App Connect is quite old now, and does not support OCP 4.10.  v4.2 is the latest version that offers support for all levels of OCP required in MAS.

### CODE REVIEW

File: ibm/mas_devops/roles/appconnect/defaults/main.yml [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-# Define where AppConnect will be installed
+# Define where AppConnect will be installed, the default values below
 appconnect_namespace: "{{ lookup('env', 'APPCONNECT_NAMESPACE') | default('ibm-app-connect', true) }}"
-# v3.0 channel is used in MAS 8.7, v2.0 channel is used in older MAS versions
-appconnect_channel: "{{ lookup('env', 'APPCONNECT_CHANNEL') | default('v3.0', true) }}"
-appconnect_dashboard_name: "{{ lookup('env', 'APPCONNECT_DASHBOARD_NAME') | default('dashboard-12020r2', true) }}"
-appconnect_license_id: "{{ lookup('env', 'APPCONNECT_DASHBOARD_NAME') | default('L-KSBM-C87FU2', true) }}"
+appconnect_channel: "{{ lookup('env', 'APPCONNECT_CHANNEL') | default('v4.2', true) }}"
+appconnect_dashboard_name: "{{ lookup('env', 'APPCONNECT_DASHBOARD_NAME') | default('dashboard-12040r2', true) }}"
+appconnect_license_id: "{{ lookup('env', 'APPCONNECT_DASHBOARD_NAME') | default('L-APEH-C9NCK6', true) }}"
```
Comment (durera) [inline ibm/mas_devops/roles/appconnect/defaults/main.yml:10]: This should be APPCONNECT_LICENSE_ID env var.

File: ibm/mas_devops/roles/appconnect/README.md [hunk 2]
```diff
@@ -14,31 +14,31 @@ Optional - Defines the targetted cluster namespace/project where AppConnect will
...
 ### appconnect_channel
-Optional - AppConnect subscription channel to be installed. For MAS 8.7, default `v3.0` channel is used. Ensure the supported dashboard version and license ID according to the AppConnect operator/channel version.
+Optional - AppConnect subscription channel to be installed.
+Ensure the supported dashboard version and license ID according to the AppConnect operator/channel version.
 
-- MAS 8.4 uses AppConnect channel subscription v1.4 and license id = L-APEH-BSVCHU
-- MAS 8.5 uses AppConnect channel subscription v2.0 and license id = L-KSBM-BZWEAT
 - MAS 8.6 uses AppConnect channel subscription v2.0 and license id = L-KSBM-C37J2R
 - MAS 8.7 uses AppConnect channel subscription v3.0 and license id = L-KSBM-C87FU2
+- MAS 8.8 uses AppConnect channel subscription v4.2 and license id = L-APEH-C9NCK6
```
Comment (durera) [inline ibm/mas_devops/roles/appconnect/README.md:22]: See comments in Slack ... do we really need to tied AppConnect channel/version 1:1 to MAS channel/version?

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/appconnect/defaults/main.yml (hunk 1 above)
Comment (andrercm) [reply ibm/mas_devops/roles/appconnect/defaults/main.yml:10]: oh that's true... glad we just go if the default most of the times, this was broken for quite some time... just fixed.

File: ibm/mas_devops/roles/appconnect/README.md (hunk 2 above)
Comment (andrercm) [reply ibm/mas_devops/roles/appconnect/README.md:22]: Waiting on response from https://ibm-watson-iot.slack.com/archives/C0195MVCEUD/p1661865932206939?thread_ts=1661842890.671729&cid=C0195MVCEUD

The default currently is using:

```
appconnect_channel: "{{ lookup('env', 'APPCONNECT_CHANNEL') | default('v4.2', true) }}"
appconnect_license_id: "{{ lookup('env', 'APPCONNECT_LICENSE_ID') | default('L-APEH-C9NCK6', true) }}"
```

Comment (andrercm) [reply ibm/mas_devops/roles/appconnect/README.md:22]: @durera can we consider it ok given the response from HPU team?

**Review by durera — APPROVED** [review]

**Pull request merged**